registry/extension/extension.go

```diff
@@ -72,27 +71,31 @@ type EnumerateExtension struct {
 var extensions map[string]InitExtensionNamespace
 var extensionsNamespaces map[string]Namespace
 
-func EnumerateRegistered(ctx context.Context) (enumeratedExtensions []EnumerateExtension) {
+func EnumerateRegistered(ctx Context) (enumeratedExtensions []EnumerateExtension) {
 	for _, namespace := range extensionsNamespaces {
 		enumerateExtension := EnumerateExtension{
 			Name:        namespace.GetNamespaceName(),
 			Url:         namespace.GetNamespaceUrl(),
 			Description: namespace.GetNamespaceDescription(),
+			Endpoints:   []string{},
 		}
 
-		registryScoped := namespace.GetRegistryRoutes()
-		for _, regScoped := range registryScoped {
-			path := fmt.Sprintf("_%s/%s/%s", regScoped.Namespace, regScoped.Extension, regScoped.Component)
-			enumerateExtension.Endpoints = append(enumerateExtension.Endpoints, path)
+		scopedRoutes := namespace.GetRepositoryRoutes()
+
+		// if the repository is not set in the context, scope is registry wide
+		if ctx.Repository == nil {
+			scopedRoutes = namespace.GetRegistryRoutes()
 		}
 
-		repositoryScoped := namespace.GetRepositoryRoutes()
-		for _, repScoped := range repositoryScoped {
-			path := fmt.Sprintf("_%s/%s/%s", repScoped.Namespace, repScoped.Extension, repScoped.Component)
+		for _, route := range scopedRoutes {
+			path := fmt.Sprintf("_%s/%s/%s", route.Namespace, route.Extension, route.Component)
 			enumerateExtension.Endpoints = append(enumerateExtension.Endpoints, path)
 		}
 
-		enumeratedExtensions = append(enumeratedExtensions, enumerateExtension)
+		// add extension to list if endpoints exist
+		if len(enumerateExtension.Endpoints) > 0 {
```

Comment:
Just looking for some context on the changes to this file.
1. Previously, both repository and registry routes were returned but now it seems to be one of the two?
2. Why are extensions added only if endpoints exist - when would that not be the case?

Reply:
1. Yes that's correct. So previously, both routes were returned but the subset of routes we want to return depends on if the discover request is at the registry or repository level. The distinction is now being made at the `EnumerateRegistered` level. If the Repository field in the context is nil, then we know that we are operating at the registry level and only want to return registry level routes for that extension namespace. 
2. It's possible that for an extension namespace, that there doesn't exist any routes at the scoped registry or repository level. In that case, we wouldn't want to add the `enumerateExtension` object to the list since no endpoints were added. The referrers extension would be an example: referrers only operates at the repository level but if we are enumerating at the registry level, then the `scopedRoutes` would be empty and we wouldn't want to add to the `enumeratedExtensions` list.